sklearn/impute/tests/test_impute.py

```diff
@@ -1383,3 +1383,20 @@ def test_imputation_order(order, idx_order):
                                random_state=0).fit(X)
         idx = [x.feat_idx for x in trs.imputation_sequence_]
         assert idx == idx_order
+
+
+def test_simple_imputation_inverse_transform():
+    # Test inverse_transform feature for np.nan
+    X = np.array([
+        [-1, np.nan, 3, -1],
+        [4, -1, 5, -1],
+        [6, 7, np.nan, -1],
+        [8, 9, 0, np.nan]
+    ])
+
+    imputer = SimpleImputer(missing_values=np.nan, strategy='mean',
+                            add_indicator=True)
+    X_trans = imputer.fit_transform(X)
```

Comment:
Added a test case with `fit` and `transform` performed separately on a different array X_2. Because imputer that is `fit` on `X_1` is only aware of the missing features of `X_1`, and would not `inverse_transform` a different array `X_2`. So I had to `fit` a new imputer for `X_2`.

Reply:
So there are two cases for the `transform` on new data:
- either the missing value is in a feature that had missing values in the fitting data. In this case, the missing indicator makes the `inverse_transform` possible.
- either the missing value is in a feature that had no missing values in the fitting data. In this case, there is no missing indicator for this feature, and the `inverse_transform` is not possible.

I would had a test for the first case, reusing `impute_1`. To make the test pass, you will need to remove the missing value in the first column of `X_2`.
For the second case, I would add a comment in the docstring of `inverse_transform` explaining the limitation.

Reply:
Thanks for the clarity! I made the following changes -
1. Modified the test case to reuse the imputer for `X_2`. 
2. Added a docstring to `inverse_transform` to highlight this limitation. I tried to frame it as clearly as I could, let me know if it can be better.